service/history/timerQueueProcessorBase.go

```diff
@@ -508,8 +532,10 @@ func (t *timerQueueProcessorBase) archiveWorkflow(
 	if err := t.deleteWorkflowExecution(task); err != nil {
 		return err
 	}
-	if resp.ArchivedInline {
-		t.metricsClient.IncCounter(metrics.HistoryProcessDeleteHistoryEventScope, metrics.WorkflowCleanupDeleteHistoryInlineCount)
+	if !archiveHistory || resp.HistoryArchivedInline {
```

Comment:
Can you include a comment explaining that visibility delete happens differently from history and thats why we do not see this same bit of code for visibility?

Reply:
Done.